stl/inc/format

```diff
@@ -240,6 +244,69 @@ constexpr const _CharT* _Parse_precision(const _CharT* _Begin, const _CharT* _En
     return _Begin;
 }
 
+template <class _CharT, _Parse_spec_callbacks<_CharT> _Callbacks_type>
+constexpr const _CharT* _Parse_format_specs(const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+    if (_Begin == _End || *_Begin == '}') {
+        return _Begin;
+    }
+
+    _Begin = _Parse_align(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    switch (*_Begin) {
+    case '+':
+        _Callbacks._On_sign(_Sign::_Plus);
+        ++_Begin;
+        break;
+    case '-':
+        _Callbacks._On_sign(_Sign::_Minus);
+        ++_Begin;
+        break;
+    case ' ':
+        _Callbacks._On_sign(_Sign::_Space);
+        ++_Begin;
+        break;
+    }
+
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    // Parse hash.
+    if (*_Begin == '#') {
+        _Callbacks._On_hash();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    // Parse zero flag
+    if (*_Begin == '0') {
+        _Callbacks._On_zero();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    _Begin = _Parse_width(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    // Parse precision
+    if (*_Begin == '.') {
+        _Begin = _Parse_precision(_Begin, _End, _Callbacks);
+    }
+
+    // Parse type
+    if (_Begin != _End && *_Begin != '}') {
+        _Callbacks._On_type(*_Begin++);

```

Comment:
This notably doesn't validate that the character is "one of a A b B c d e E f F g G o p s x X" per [format.string.id]/1. Is this an oversight, or do you expect to validate the type elsewhere later?

Reply:
validated later (in the implementation of on_type. It may be more consistent to validate it here, but libfmt does it in on_type, and tbh I'm more comfortable being inconsistent but close to libfmt than being consistent and far away.

Reply:
That's fine - I just wanted to be sure it was intentional.